app/Http/Controllers/ClientController.php

```diff
@@ -22,6 +25,83 @@ public function index()
         return view('website.frontend.index', compact('posts', 'category'));
     }
 
+    public function filterCategory($id)
+    {
+        $posts = Post::where('category_id', $id)->with('category')->get();
+        $latestPost = Post::orderBy('created_at', 'desc')->where('category_id', $id)->take(1)->get();
+        $allCategory = Category::where('parent_id', config('number_format.parent_id'))->get();
+        $allCategory->load('children');
+
+        return view('website.frontend.filter_category', compact('posts', 'latestPost', 'allCategory'));
+    }
+
+    public function postLike(Request $request)
+    {
+        $data = $request->all();
+        $post_id = $request['post_id'];
```

Comment:
```suggestion
        $postId = $request['post_id'];
```
lỗi này lanaf nữa là phạt nhé